packages/react-components/react-timepicker-compat-preview/src/components/TimePicker/TimePicker.test.tsx

```diff
@@ -42,16 +42,38 @@ describe('TimePicker', () => {
     );
     handleTimeSelect.mockClear();
 
-    // Do not call onTimeSelect when Tab out but the value remains the same
+    // Do not call onTimeSelect on Enter when the value remains the same
     fireEvent.keyDown(input, { key: 'Enter', code: 'Enter' });
     expect(handleTimeSelect).toHaveBeenCalledTimes(0);
 
-    // Call onTimeSelect when Tab out and the value changes
+    // Call onTimeSelect on Enter when the value changes
     userEvent.type(input, '111{enter}');
     expect(handleTimeSelect).toHaveBeenCalledTimes(1);
     expect(handleTimeSelect).toHaveBeenCalledWith(
       expect.anything(),
       expect.objectContaining({ selectedTimeText: '10:30111', error: 'invalid-input' }),
     );
   });
+
+  it('when freeform, trigger onTimeSelect on blur when value change', () => {
```

Comment:
IMO JSDOM tests are generally not a great idea for anything focus related, browser tests will work best there

Reply:
Noted 👍 I plan to have e2e test for the same scenario as well.